CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/grid/ArrangeInGridHorizontalLineAction.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2010-2020 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.arrangements.grid;
+
+import au.gov.asd.tac.constellation.plugins.arrangements.ArrangementPluginRegistry;
+import au.gov.asd.tac.constellation.graph.interaction.InteractiveGraphPluginRegistry;
+import au.gov.asd.tac.constellation.graph.node.GraphNode;
+import au.gov.asd.tac.constellation.plugins.PluginExecutor;
+import java.awt.event.ActionEvent;
+import javax.swing.AbstractAction;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.NbBundle.Messages;
+
+/**
+ * Arranging components in a grid of grids.
+ *
+ * @author serpens24
+ */
+@ActionID(category = "Arrange", id = "au.gov.asd.tac.constellation.plugins.arrangements.grid.ArrangeInGridHorizontalLineAction")
+@ActionRegistration(displayName = "#CTL_ArrangeInGridHorizontalLineAction", iconBase = "au/gov/asd/tac/constellation/plugins/arrangements/grid/resources/gridhoriz.png", surviveFocusChange = true)
+@ActionReferences({
+    @ActionReference(path = "Menu/Arrange", position = 30),
+    @ActionReference(path = "Toolbars/Arrange", position = 30),
+    @ActionReference(path = "Shortcuts", name = "C-A-H")
+})
+@Messages("CTL_ArrangeInGridHorizontalLineAction=Horizontal Line")
+public final class ArrangeInGridHorizontalLineAction extends AbstractAction {
+    
+    
+    private final GraphNode context;
+
+    /**
+     * Construct a new instance.
+     *
+     * @param context GraphNode context.
+     */
+    public ArrangeInGridHorizontalLineAction(final GraphNode context) {
+        this.context = context;
+    }
+
+    
+    @Override
+    public void actionPerformed(final ActionEvent e) {
+        PluginExecutor.startWith(ArrangementPluginRegistry.GRID_GENERAL)
+                .set(ArrangeInGridGeneralPlugin.GRID_CHOICE_PARAMETER_ID, "Horizontal line")
```

Comment:
Can you please use the `GridChoice` enums instead of the String?

Reply:
Hi, I will make these changes. I wasnt aware of the GridChoice enums.